framework/include/auxkernels/MaterialRealVectorValueAux.h

```diff
@@ -13,15 +13,19 @@
 #include "MaterialAuxBase.h"
 
 // Forward declarations
-class MaterialRealVectorValueAux;
+template <bool>
+class MaterialRealVectorValueAuxTempl;
+typedef MaterialRealVectorValueAuxTempl<false> MaterialRealVectorValueAux;
+typedef MaterialRealVectorValueAuxTempl<true> ADMaterialRealVectorValueAux;
```

Comment:
Yes since you cannot mix `getADMaterialProperty` and `getMaterialProperty` we need to different AuxKernel versions depending on which we want to retrieve